### PR TITLE
Style Guide Update

### PR DESCRIPTION
**Pull request by JensHaley**
**What this PR does / why we need it**:

Updates the contribution style guide.
Information about HUGO specific frontmatter (in the versioning section) was removed - I believe it's better to keep this separately.

### CODE REVIEW

File: website/documentation/contribute/20_documentation/20_style/_index.md [hunk 1]
```diff
@@ -7,155 +7,144 @@ This page gives writing style guidelines for the Gardener documentation.
...
+leads to an awkward construction.
 
-### Use code style for inline code and commands
+|  Do  | Don't |
+|:---|:---|
+| The `Pod` has two containers.   | The pod has two containers.    |
+| The `Deployment` is responsible for ...   |  The Deployment object is responsible for ...    |
+| A `PodList` is a list of Pods. | A Pod List is a list of pods.  |
+| The `gardener-control-manager` has control loops... | The gardener-control-manager has control loops...|
+| The gardenlet starts up with a bootstrap `kubeconfig` having a bootstrap token that allows to create `CertificateSigningRequest` (CSR) resources. | The gardenlet starts up with a bootstrap kubeconfig having a bootstrap token that allows to create CertificateSigningRequest (CSR) resources. |
```
Comment (rfranzke) [inline website/documentation/contribute/20_documentation/20_style/_index.md:86]:
```suggestion
| The `gardenlet` starts up with a bootstrap `kubeconfig` having a bootstrap token that allows to create `CertificateSigningRequest` (CSR) resources. | The gardenlet starts up with a bootstrap kubeconfig having a bootstrap token that allows to create CertificateSigningRequest (CSR) resources. |
```

**Comment by JensHaley** [comment]
@rfranzke : I included your correction and this PR has now been open for a while without anybody objecting to my changes.

Do you think we can merge this now?

**Comment by rfranzke** [comment]
I don't have objections.

**Review by g-pavlov — APPROVED** [review]
\lgtm

**Pull request merged**